middleware/src/rpcserver/rpcserver.go

```diff
@@ -83,73 +89,112 @@ func NewRPCServer(middleware Middleware) *RPCServer {
 }
 
 // GetSystemEnv sends the middleware's GetEnvResponse over rpc
-func (server *RPCServer) GetSystemEnv(args int, reply *rpcmessages.GetEnvResponse) error {
+func (server *RPCServer) GetSystemEnv(dummyArg bool, reply *rpcmessages.GetEnvResponse) error {
```

Comment:
TODO for the future: since the rpc server requires args and returning an error, we should write a wrapper so we don't need to pass `dummyArg` and `return nil` everywhere